File: src/main/java/com/yunus/entity/Note.java
package com.yunus.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "notes")
@NoArgsConstructor
@AllArgsConstructor
public class Note extends BaseEntity{

    @Column(nullable = false,columnDefinition = "TEXT")
    private String content;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id")
    private Customer customer;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "conversation_id")
    private Conversation conversation;

    @ManyToOne(fetch = FetchType.LAZY,optional = false)
    @JoinColumn(name = "created_by_id",nullable = false)
    private User createdBy;



}

File: Frontend/src/types/note.ts
export type NoteResponse = {
  id: string;
  customerId?: string | null;
  content: string;
  conversationId?: string | null;
  createdById: string;
  createdByFullName: string;
  createdAt: string;
  updatedAt: string;
};

export type NoteCreateRequest = {
  content: string;
};

export type NoteUpdateRequest = {
  content: string;
};
